freedomofpress/qubes-template-securedrop-workstation
# Adds new signing key to apt repo bootstrap logic

Follow-up to #21, and a crucial requirement for #20.
Ensures that the newest signing key is inserted early in the
VM creation. It's not strictly necessary for the next template build,
but will be required for all template builds after 2021-06-30, when
the old signing key is set to expire.

### Testing

First, visual review:

1. Confirm that the two (2) pubkey files in this repo corresponding to the old/new fingerprints described in https://github.com/freedomofpress/securedrop/issues/5923
2. Inspect the scripts and confirm the changes look well structured (i.e, both keys are being added for the apt repo)

Then, a bit of functional verification, to ensure nothing broke:
1. Check out this branch
2. Apply the patch below, to trust my individual key, for the purposes of review
3. `make template`

Should work just fine. 

Also, confirm the shellcheck linting is happy:

1. `shellcheck build-workstation-template` shows no errors
2. `shellcheck securedrop-workstation/04_install_qubes_post.sh` shows no errors


<details>
<summary>
git patch for review
</summary>

```diff
--- a/build-workstation-template
+++ b/build-workstation-template
@@ -45,7 +45,8 @@ mkdir -m 700 "${gpg_homedir_sdw}"
 # See README regarding feature branches for testing changes.
 release_key_fingerprint_2020="22245C81E3BAEB4138B36061310F561200F4AD77"
 release_key_fingerprint_2021="2359E6538C0613E652955E6C188EDD3B7B22E6A3"
-for k in "$release_key_fingerprint_2020" "$release_key_fingerprint_2021"; do
+conor_key="0B095DF428491E147B615CD3F08893B959CAB065"
+for k in "$release_key_fingerprint_2020" "$release_key_fingerprint_2021" "$conor_key"; do
     for d in "$gpg_homedir" "$gpg_homedir_sdw"; do
         gpg --homedir "$d" --keyserver keys.openpgp.org --recv-key "$k"
         echo "${k}:6:" | gpg --homedir "$d" --import-ownertrust
diff --git a/securedrop-workstation.conf b/securedrop-workstation.conf
index 7554999..bbd45d1 100644
--- a/securedrop-workstation.conf
+++ b/securedrop-workstation.conf
@@ -8,7 +8,7 @@
 
 GIT_URL_template_securedrop_workstation = https://github.com/freedomofpress/qubes-template-securedrop-workstation
 
-BRANCH_template_securedrop_workstation ?= main
+BRANCH_template_securedrop_workstation ?= add-new-signing-key-in-build-logic
 BUILDER_PLUGINS += builder-debian
 BUILDER_PLUGINS += template-securedrop-workstation
 USE_QUBES_REPO_VERSION = 4.0
```
